### PR TITLE
Add feature update cards for 2.9 and 2.10

### PR DESCRIPTION
**Pull request by BenWilson2**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/BenWilson2/mlflow/pull/10841?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/10841/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 10841
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

Adds feature update cards for 2.9 and 2.10 releases

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [X] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [X] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [X] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [X] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: docs/source/_static/images/logos/gateway-header-image.png
Comment (harupy) [inline docs/source/_static/images/logos/gateway-header-image.png:1]: is it possible to reduce the image size? 1.38 MB is not small

Comment (BenWilson2) [reply docs/source/_static/images/logos/gateway-header-image.png:1]: I think I can with Photoshop pretty easily :) I'll reduce it after retro

File: docs/source/new-features/index.rst [hunk 1]
```diff
@@ -9,6 +9,78 @@ Find out about the details of major features, changes, and deprecations below.
 
     <section>
         <article class="new-content-grid">
+            <div class="grid-card">
+                <div class="content-container">
+                    <div class="header">
+                        MLflow Site Overhaul 
+                    </div>
+                    <img class="card-image" src="../_static/images/logos/mlflow-logo.svg" alt="MLflow"></img>
```
Comment (B-Step62) [inline docs/source/new-features/index.rst:17]: Can we add screenshot for the new home page?

Comment (BenWilson2) [reply docs/source/new-features/index.rst]: Great idea! Added!

File: docs/source/_static/images/logos/gateway-header-image.png
Comment (BenWilson2) [reply docs/source/_static/images/logos/gateway-header-image.png:1]: reduced it to ~ 10x smaller

**Review by B-Step62 — APPROVED** [review]
LGTM!!

**Review by WeichenXu123 — APPROVED** [review]

**Pull request merged**